examples/c/uprobe.c

```diff
@@ -96,24 +101,31 @@ int main(int argc, char **argv)
 	 * processes that use the same binary executable; to do that we need
 	 * to specify -1 as PID, as we do here
 	 */
-	skel->links.uretprobe = bpf_program__attach_uprobe(skel->progs.uretprobe,
-							   true /* uretprobe */,
-							   -1 /* any pid */,
-							   "/proc/self/exe",
-							   uprobe_offset);
-	if (!skel->links.uretprobe) {
+	skel->links.uretprobe_add = bpf_program__attach_uprobe(skel->progs.uretprobe_add,
+							       true /* uretprobe */,
+							       -1 /* any pid */,
+							       "/proc/self/exe",
+							       uprobe_offset);
+	if (!skel->links.uretprobe_add) {
 		err = -errno;
 		fprintf(stderr, "Failed to attach uprobe: %d\n", err);
 		goto cleanup;
 	}
 
+	err = uprobe_bpf__attach(skel);
```

Comment:
how about instead showing off generic per-program attachment here:

```
skel_.links.uprobe_sub = bpf_program__attach(skel->progs.uprobe_add);
```
and the same for retprobe (with error checking, of course).

I think a lot of newbies will glance over uprobe_bpf__attach() and won't realize what's going on and where and when uprobe attachment is actually happening.

WDYT?

Reply:
The `uprobe_bpf__attach` is to demostrate auto-attach, just like the USDT example.